src/mod_auth_cas.c

```diff
@@ -2081,6 +2113,16 @@ int cas_authenticate(request_rec *r)
 	ticket = getCASTicket(r);
 	cookieString = getCASCookie(r, (ssl ? d->CASSecureCookie : d->CASCookie));
 
+	// prevent infinite redirect loops by allowing subsequent authentication responses to pass through, leaving the ticket parameter intact
+	if(c->CASAllowSubAuth && (ticket != NULL) && (cookieString != NULL) && ap_is_initial_req(r) && isValidCASCookie(r, c, cookieString, &remoteUser, &attrs)) {
+		cas_set_attributes(r, attrs);
+		r->user = remoteUser;
```

Comment:
Should we check that remoteUser exists here?

Reply:
Doesn't seem to have an affect either way during basic testing, but I added the check to match the other code.

Reply:
I'm more concerned when a remoteUser doesn't exist.

I wonder if this should mirror https://github.com/portlandcc/mod_auth_cas/blob/allow_sub_auth/src/mod_auth_cas.c#L2236.

If remoteUser could be NULL, I don't think returning OK is the right thing to do. I haven't done any analysis, but it is needed.

Reply:
If the remoteUser is NULL, something went wrong reading or writing the cookie, which seems unlikely. If we want to be super safe, it's probably not a bad idea to add it to the outer if.

Reply:
The remoteUser NULL check was moved to the outer if.